src/pages/[locale]/[...path]/index.astro

```diff
@@ -55,7 +48,7 @@ type Params = {
 const { locale, path } = Astro.params as Params;
 const variables = { locale, slug: getPageSlugFromPath(path) };
 const { page } = (await datocmsRequest<PageQuery>({ query, variables })) as {
-  page: PageRecord;
+  page: NonNullable<PageQuery['page']>; // Only NonNullable when statically generated. Handle as a 404 when this is a server route!
```

Comment:
I think we can now solve this a different way. Since v4.13 [Astro supports rewrites, including a `404` pattern](https://docs.astro.build/en/guides/routing/#rewrites). So we should be able to do this instead:

```ts
const { page } = await datocmsRequest<PageQuery>({ query, variables })

if (!page) {
  return Astro.rewrite("/404")
}
```
I haven't used this before and haven't tested this yet either :)

Reply:
Yes this is indeed what I would prefer too, and what I mentioned on Slack. But I would argue that the better implementation would be to have a function do this, either by wrapping the datocmsRequest, or reimplementing the datocmsRequest, or extending the datocmsRequest. I would prefer to park this for another PR. Here I just wanted to make sure that the type is correct, and does not throw for the routing functions.